fractal_server/app/routes/api/v2/task_collection.py

```diff
@@ -119,8 +121,13 @@ async def collect_tasks_pip(
     except FileExistsError:
         venv_path = create_package_dir_pip(task_pkg=task_pkg, create=False)
         try:
-            task_collect_status = get_collection_data(venv_path)
-            for task in task_collect_status.task_list:
+            package_path = get_absolute_venv_path(venv_path)
+            collection_path = get_collection_path(package_path)
+            with collection_path.open() as f:
+                task_collect_status = json.load(f)
+            for task in [
+                TaskReadV2(**task) for task in task_collect_status["task_list"]
```

Comment:
We should make sure that this line does not raise a KeyError, but a 422 (in case `task_list` is missing)